ncclient/transport/ssh.py

```diff
@@ -46,6 +48,18 @@
 
 TICK = 0.1
 
+#
+# Define delimiters for chunks and messages for netconf 1.1 chunk enoding.
+# When matched:
+#
+# * result.group(0) will contain whole matched string
+# * result.group(1) will contain the '#[0-9]+'
+# * result.group(2) will contain the digit string for a chunk
+# * result.group(3) will be defined if '##' found
+#
+RE_NC11_DELIM = re.compile('\n(#([0-9]+)|(##))\n')
```

Comment:
Suggest changing the outer parens to a non-capturing grouping since you're not using it as a match group anyway:

```python
# * result.group(0) will contain whole matched string
# * result.group(1) will contain the digit string for a chunk
# * result.group(2) will be defined if '##' found
RE_NC11_DELIM = re.compile(r'\n(?:#([0-9]+)|(##))\n')
```

Reply:
Will adopt this modified regex.